packages/@aws-cdk/aws-stepfunctions-tasks/lib/batch/submit-job.ts

```diff
@@ -0,0 +1,312 @@
+   * @default - No parameters are passed
+   */
+  readonly payload?: sfn.TaskInput;
+
+  /**
+   * The number of times to move a job to the RUNNABLE status.
+   * You may specify between 1 and 10 attempts.
+   * If the value of attempts is greater than one,
+   * the job is retried on failure the same number of attempts as the value.
+   *
+   * @default 1
+   */
+  readonly attempts?: number;
+}
+
+/**
+ * Task to submits an AWS Batch job from a job definition.
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-batch.html
+ */
+export class BatchSubmitJob extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: BatchSubmitJobProps) {
+    super(scope, id, props);
+
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.RUN_JOB;
+    validatePatternSupported(this.integrationPattern, BatchSubmitJob.SUPPORTED_INTEGRATION_PATTERNS);
+
+    // validate arraySize limits
+    withResolved(props.arraySize, (arraySize) => {
+      if (arraySize !== undefined && (arraySize < 2 || arraySize > 10_000)) {
+        throw new Error(`arraySize must be between 2 and 10,000. Received ${arraySize}.`);
+      }
+    });
+
+    // validate dependency size
+    if (props.dependsOn && props.dependsOn.length > 20) {
+      throw new Error(`dependencies must be 20 or less. Received ${props.dependsOn.length}.`);
+    }
+
+    // validate attempts
+    withResolved(props.attempts, (attempts) => {
+      if (attempts !== undefined && (attempts < 1 || attempts > 10)) {
+        throw new Error(`attempts must be between 1 and 10. Received ${attempts}.`);
+      }
+    });
+
+    // validate timeout
+    // tslint:disable-next-line:no-unused-expression
+    props.timeout !== undefined && withResolved(props.timeout.toSeconds(), (timeout) => {
+      if (timeout < 60) {
+        throw new Error(`attempt duration must be greater than 60 seconds. Received ${timeout} seconds.`);
+      }
+    });
+
+    // This is required since environment variables must not start with AWS_BATCH;
+    // this naming convention is reserved for variables that are set by the AWS Batch service.
+    if (props.containerOverrides?.environment) {
+      Object.keys(props.containerOverrides.environment).forEach(key => {
+        if (key.match(/^AWS_BATCH/)) {
+          throw new Error(
+            `Invalid environment variable name: ${key}. Environment variable names starting with 'AWS_BATCH' are reserved.`,
+          );
+        }
+      });
+    }
+
+    this.taskPolicies = this.configurePolicyStatements();
+  }
+
+  protected renderTask(): any {
```

Comment:
Avoid using `any` on a public API, except for very specific use cases. Is there a stronger type we can assign to this?

Why is this protected? I can't find this used anywhere in this PR.

Reply:
Whoops. I just realized `renderTask()` is actually coming from the Task base class. Ignore.